File: java/PlanesAndroid/app/src/main/java/com/planes/android/PlanesRoundInterface.java
package com.planes.android;

//TODO: to move in common
public interface PlanesRoundInterface {

    //creates the PlaneRound object in the game engine
    //must be called a single time
    void createPlanesRound();

    //show the planes
    int getRowNo();
    int getColNo();
    int getPlaneNo();
    int getPlaneSquareType(int i, int j, int isComputer);

    //edit the board
    int movePlaneLeft(int idx);
    int movePlaneRight(int idx);
    int movePlaneUpwards(int idx);
    int movePlaneDownwards(int idx);
    int rotatePlane(int idx);
    void doneClicked();

    //play the game
    int playerGuessAlreadyMade(int row, int col);
    void playerGuess(int row, int col);
    boolean playerGuess_RoundEnds();
    boolean playerGuess_IsPlayerWinner();
    boolean playerGuess_IsDraw();
    boolean playerGuess_ComputerMoveGenerated();
    int playerGuess_StatNoPlayerMoves();
    int playerGuess_StatNoPlayerHits();
    int playerGuess_StatNoPlayerMisses();
    int playerGuess_StatNoPlayerDead();
    int playerGuess_StatNoPlayerWins();
    int playerGuess_StatNoComputerMoves();
    int playerGuess_StatNoComputerHits();
    int playerGuess_StatNoComputerMisses();
    int playerGuess_StatNoComputerDead();
    int playerGuess_StatNoComputerWins();
    int playerGuess_StatNoDraws();

    void roundEnds();
    void initRound();

    //show the guesses
    int getPlayerGuessesNo();
    int getPlayerGuessRow(int idx);
    int getPlayerGuessCol(int idx);
    int getPlayerGuessType(int idx);

    int getComputerGuessesNo();
    int getComputerGuessRow(int idx);
    int getComputerGuessCol(int idx);
    int getComputerGuessType(int idx);

    int getGameStage();
}

File: kotlin/PlanesAndroid/app/src/main/java/com/planes_multiplayer/android/PlanesRoundInterface.kt
package com.planes.android

//TODO: to move in common
interface PlanesRoundInterface {
    //creates the PlaneRound object in the game engine
    //must be called a single time
    fun createPlanesRound()

    //show the planes
    fun getRowNo(): Int
    fun getColNo(): Int
    fun getPlaneNo(): Int
    fun getPlaneSquareType(i: Int, j: Int, isComputer: Int): Int

    //edit the board
    fun movePlaneLeft(idx: Int): Int
    fun movePlaneRight(idx: Int): Int
    fun movePlaneUpwards(idx: Int): Int
    fun movePlaneDownwards(idx: Int): Int
    fun rotatePlane(idx: Int): Int
    fun doneClicked()

    //play the game
    fun playerGuessAlreadyMade(row: Int, col: Int): Int
    fun playerGuess(row: Int, col: Int)
    fun playerGuess_RoundEnds(): Boolean
    fun playerGuess_IsPlayerWinner(): Boolean
    fun playerGuess_IsDraw(): Boolean
    fun playerGuess_ComputerMoveGenerated(): Boolean
    fun playerGuess_StatNoPlayerMoves(): Int
    fun playerGuess_StatNoPlayerHits(): Int
    fun playerGuess_StatNoPlayerMisses(): Int
    fun playerGuess_StatNoPlayerDead(): Int
    fun playerGuess_StatNoPlayerWins(): Int
    fun playerGuess_StatNoComputerMoves(): Int
    fun playerGuess_StatNoComputerHits(): Int
    fun playerGuess_StatNoComputerMisses(): Int
    fun playerGuess_StatNoComputerDead(): Int
    fun playerGuess_StatNoComputerWins(): Int
    fun playerGuess_StatNoDraws(): Int
    fun roundEnds()
    fun initRound()

    //show the guesses
    fun getPlayerGuessesNo(): Int
    fun getPlayerGuessRow(idx: Int): Int
    fun getPlayerGuessCol(idx: Int): Int
    fun getPlayerGuessType(idx: Int): Int
    fun getComputerGuessesNo(): Int
    fun getComputerGuessRow(idx: Int): Int
    fun getComputerGuessCol(idx: Int): Int
    fun getComputerGuessType(idx: Int): Int
    fun getGameStage(): Int

    //game options
    fun setComputerSkill(skill: Int): Boolean
    fun setShowPlaneAfterKill(show: Boolean): Boolean
    fun getComputerSkill(): Int
    fun getShowPlaneAfterKill(): Boolean

}